src/overlays/actors/ovl_En_Pamera/z_en_pamera.c

```diff
@@ -22,135 +104,791 @@ const ActorInit En_Pamera_InitVars = {
+                    gSaveContext.weekEventReg[59] &= (u8)~1;
+                }
+                func_80BD8700(this);
+            } else {
+                this->pathIndex = this->pathPointsCount - 1;
+                Math_Vec3s_ToVec3f(&this->actor.world.pos, &this->pathPoints[this->pathIndex]);
+                Math_Vec3s_ToVec3f(&sp44, &this->pathPoints[this->pathIndex - 1]);
+                this->actor.world.rot.y = this->actor.shape.rot.y = Math_Vec3f_Yaw(&sp44, &this->actor.world.pos);
+                func_80BD8CCC(this);
+            }
+        } else {
+            gSaveContext.weekEventReg[59] |= 1;
+            func_80BD8FF0(this);
+        }
+    }
+}
+
+s32 func_80BD84F0(EnPamera* this, GlobalContext* globalCtx) {
+    Actor* actor = globalCtx->actorCtx.actorList[ACTORCAT_DOOR].first;
+
+    while (actor != NULL) {
+        if ((actor->id == ACTOR_EN_DOOR) && (Math_Vec3f_DistXZ(&this->actor.world.pos, &actor->world.pos) < 200.0f)) {
+            this->actor.child = actor;
+            return 1;
+        }
+        actor = actor->next;
+    }
+    this->actor.child = NULL;
+    return 0;
+}
+
+void func_80BD8588(EnPamera* this, GlobalContext* globalCtx) {
+    Path* path = &globalCtx->setupPathList[((this->actor.params & 0xFF0) >> 4)];
+    Vec3f sp28;
+
+    if (path == 0) {
+        Actor_MarkForDeath(&this->actor);
+    }
+    if (gSaveContext.weekEventReg[61] & 4) {
+        path = &globalCtx->setupPathList[path->unk1];
+    }
+    this->pathPoints = Lib_SegmentedToVirtual(path->points);
+    this->pathIndex = 0;
+    this->pathPointsCount = path->count;
+    this->pathID = path->unk1;
+    Math_Vec3s_ToVec3f(&sp28, this->pathPoints);
+    this->actor.world.rot.y = this->actor.shape.rot.y = Math_Vec3f_Yaw(&this->actor.world.pos, &sp28);
+}
+
+void func_80BD8658(EnPamera* this) {
+    s16 nextCutscene;
+    s32 i;
+
+    nextCutscene = this->actor.cutscene;
+    for (i = 0; i < ARRAY_COUNT(this->cutscenes); i++) {
+        if (nextCutscene == -1) {
+            break;
+        }
+        this->cutscenes[i] = nextCutscene;
+        nextCutscene = ActorCutscene_GetAdditionalCutscene(nextCutscene);
+    }
+}
+
+void EnPamera_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnPamera* this = THIS;
+
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+}
+
+void func_80BD8700(EnPamera* this) {
+    this->unk_31C = 0;
+    this->actor.flags &= ~1;
+    func_800BDC5C(&this->skelAnime, animations, 0);
+    this->actionFunc = func_80BD8758;
+}
+
+void func_80BD8758(EnPamera* this, GlobalContext* globalCtx) {
+    if (this->unk_31C++ >= 0x709) {
+        if (ActorCutscene_GetCanPlayNext(this->cutscenes[0]) && (this->cutscenes[0] != -1)) {
+            ActorCutscene_StartAndSetUnkLinkFields(this->cutscenes[0], &this->actor);
+            func_800E02AC(Play_GetCamera(globalCtx, ActorCutscene_GetCurrentCamera(this->cutscenes[0])), this);
```

Comment:
Now this is interesting. I would assume you need `&this->actor` and it didn't raise an error because this function hasn't been declared. I have this function decomped in camera, but it seemed like the second argument was player: 
`void func_800E02AC(Camera* camera, Player* player);`
https://github.com/engineer124/mm/blob/3aa60135bfc1b1f83b7f95b93a10a6f14eb5811c/src/code/z_camera.c#L7883-L7893
This is definitely the function when Pamela gets startled so the camera is set to be fixed on Pamela, which means I may need to reinterpret the Camera side of things. It probably just takes in an actor, so for now just put in the function declaration:
`void func_800E02AC(Camera* camera, Actor* actor);`
and put in `&this->actor` for the two uses of this function in this file